kore/test/Test/Kore/Step/Simplification/Integration.hs

```diff
@@ -191,6 +196,166 @@ test_simplificationIntegration =
+                        ]
+                    )
+                )
+                Conditional
+                    { term = mkExists Mock.x (mkElemVar Mock.x)
+                    , predicate = requirement
+                    , substitution = mempty
+                    }
+        assertEqual "" expect actual
+    , testCase "simplification with top predicate (set variable capture)" $ do
+        let requirement =
+                makeEqualsPredicate (Mock.f (mkSetVar Mock.setX)) (Mock.g Mock.b)
+            expect =
+                OrPattern.fromPatterns
+                [ Conditional
+                    { term = mkNu Mock.var_setX_0 (mkSetVar Mock.var_setX_0)
+                    , predicate = requirement
+                    , substitution = mempty
+                    }
+                ]
+        actual <-
+            evaluateWithAxioms
+                ( axiomPatternsToEvaluators
+                    ( Map.fromList
+                        [ (AxiomIdentifier.Application Mock.functionalConstr10Id
+                          , [ axiom
+                                (Mock.functionalConstr10 (mkElemVar Mock.x))
+                                (Mock.g Mock.a)
+                                requirement
+                            ]
+                          )
+                        ]
+                    )
+                )
+                Conditional
+                    { term = mkNu Mock.setX (mkSetVar Mock.setX)
+                    , predicate = requirement
+                    , substitution = mempty
+                    }
+        assertEqual "" expect actual
+    , testCase "simplification with top predicate (set variable capture)" $ do
+        let requirement =
+                makeEqualsPredicate (Mock.f (mkSetVar Mock.setX)) (Mock.g Mock.b)
+            expect =
+                OrPattern.fromPatterns
+                [ Conditional
+                    { term = mkMu Mock.var_setX_0 (mkSetVar Mock.var_setX_0)
+                    , predicate = requirement
+                    , substitution = mempty
+                    }
+                ]
+        actual <-
+            evaluateWithAxioms
+                ( axiomPatternsToEvaluators
+                    ( Map.fromList
+                        [ (AxiomIdentifier.Application Mock.functionalConstr10Id
+                          , [ axiom
+                                (Mock.functionalConstr10 (mkElemVar Mock.x))
+                                (Mock.g Mock.a)
+                                requirement
+                            ]
+                          )
+                        ]
+                    )
+                )
+                Conditional
+                    { term = mkMu Mock.setX (mkSetVar Mock.setX)
+                    , predicate = requirement
+                    , substitution = mempty
+                    }
+        assertEqual "" expect actual
+    , testCase "simplification with top predicate (variable capture)" $ do
```

Comment:
Could you move this near the similar `exists` test, and add `exists` and `forall` to their names?